pkg/cmd/pr/list/list.go

```diff
@@ -116,6 +118,19 @@ func listRun(opts *ListOptions) error {
 		return err
 	}
 
+	argv, err := shlex.Split(opts.Search)
+	if err != nil {
+		return err
+	}
+
+	re := regexp.MustCompile(`^merged:.*`)
+
+	for _, arg := range argv {
+		if opts.State == "open" && (arg == "is:closed" || arg == "is:merged" || re.MatchString(arg)) {
```

Comment:
This looks good. Let's add a few more conditions that I forgot about
- `state:closed` seems to be a synonym for `is:closed`
- ditto `state:merged`
- let's also match on the `closed:` prefix like we do with `merged:` prefix

https://docs.github.com/en/github/searching-for-information-on-github/searching-on-github/searching-issues-and-pull-requests